cromshell

```diff
@@ -282,7 +282,44 @@ function checkPrerequisites
   return 0
 }
 
-function checkForComplexHelpArgument() 
+function assertRequiredFileIsNonEmpty()
+{
+
+  local fileName=$1
+  local fileDescription=$2
+  if [[ -z ${fileName} ]]
+	then
+		error "Error: required file ${fileDescription} was not specified."
+		exit 6
+	else
+		if [[ -f ${fileName} ]]
+		then
+			if [[ -s ${fileName} ]]
+			then
+				return 0
+			else
+				error "Error: file ${fileName} is an empty file."
+				exit 6
+			fi
+		else
+			error "Error: file ${fileName} does not exist."
+			exit 6;
+		fi
+	fi
+}
+
+function assertOptionalFileIsNonEmpty()
+{
+	local fileName=$1
+	if [[ -z ${fileName} ]]
+	then
+		return 0
+	else
+		assertRequiredFileIsNonEmpty ${fileName}
```

Comment:
If you're going to call into this function you should have this also take a description string so you can pass it along and keep the user output consistent.